src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -371,6 +396,7 @@ private static void HookOrProcessResult(HttpWebRequest request)
     private static void ProcessResult(IAsyncResult asyncResult, AsyncCallback asyncCallback, Activity activity, object result, bool forceResponseCopy, HttpWebRequest request, long startTimestamp)
     {
         HttpStatusCode? httpStatusCode = null;
+        object errorType = null;
```

Comment:
Curious why not `string` for the type?

Reply:
Looks like because it may also be set via TelemetryHelper.GetBoxedStatusCode. But then we are mixing string and int values, no? Spec says it should be a string value. Do we need to expand the helper to also have a string cache?

Something like...

```csharp
internal static class TelemetryHelper
{
    public static readonly (object, string)[] BoxedStatusCodes;

    static TelemetryHelper()
    {
        BoxedStatusCodes = new (object, string)[500];
        for (int i = 0, c = 100; i < BoxedStatusCodes.Length; i++, c++)
        {
            BoxedStatusCodes[i] = (c, c.ToString());
        }
    }

    public static object GetBoxedStatusCode(int statusCode)
    {
        if (statusCode >= 100 && statusCode < 600)
        {
            return BoxedStatusCodes[statusCode - 100].Item1;
        }

        return statusCode;
    }

    public static string GetStatusCodeString(int statusCode)
    {
        if (statusCode >= 100 && statusCode < 600)
        {
            return BoxedStatusCodes[statusCode - 100].Item2;
        }

        return statusCode.ToString();
    }
}
```

Reply:
avoids cast [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/b1c8bd4fa7942116c62efaee9e496a7a7743ccff/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs#L464)

Reply:
@CodeBlanch You are right. Spec does say (represented as string). Will fix it.

Reply:
fixed

Reply:
Do you have to do this for `HttpHandlerDiagnosticListener` as well?

Reply:
Yes, I will do a separate PR for that and also for ASP.NET Core

Reply:
Correction, we don't set `error.type` to statuscode in ASP.NET Core due to https://github.com/dotnet/aspnetcore/issues/51620